File: backend/src/main/java/solid/backend/entity/CodeFile.java
package solid.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.Comment;

import java.time.LocalDate;

@Data
@Entity
@Table(name = "code_file")
public class CodeFile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "code_file_id")
    @Comment("파일 ID")
    private Integer codeFileId;

    @Column(name = "code_file_path", length = 100, nullable = false)
    @Comment("파일 경로")
    private String codeFilePath;

    @Column(name = "code_file_name",length = 20, nullable = false)
    @Comment("파일 이름")
    private String codeFileName;

    @Column(name = "code_file_upload_dt", nullable = false)
    @Comment("파일 업로드일")
    private LocalDate codeFileUploadDt;

    @Column(name = "code_file_create_dt", nullable = false)
    @Comment("파일 생성일")
    private LocalDate codeFileCreateDt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "directory_id")
    private Directory directory;

}

File: frontend/src/types/codefile.ts
export interface CodeFileDelDto {
  codeFileId: number;
}

export interface CodeFileListDto {
  codeFileId: number;
  directoryId: number;
  codeFilePath: string;
  codeFileName: string;
  codeFileUploadDt: string;
  codeFileCreateDt: string;
}

export interface CodeFileSaveDto {
  directoryId: number;
  codeFileName: string;
  codeContent: string;
}

export interface CodeFileUpdDto {
  codeFileId: number;
  codeContent: string;
}
